src/SystemCtl.php

```diff
@@ -94,7 +100,7 @@ public function listUnits(?string $unitPrefix = null, array $unitTypes = self::S
         $commands = ['list-units'];
 
         if ($unitPrefix) {
-            $commands[] = [$unitPrefix . '*'];
+            $commands[] = $unitPrefix . '*';
```

Comment:
This is the complexity added by the variadic argument.
You have to thing back and forth to get it all in your mind where you could just use a simple array or another way of aggregation (e.g. `add` method) ;)